src/shared/components/lazyMobXTable/LazyMobXTable.tsx

```diff
@@ -134,6 +134,35 @@ export function lazyMobXTableSort<T>(data:T[], metric:SortMetric<T>, ascending:b
     return dataAndValue.map(x=>x.data);
 }
 
+function getListOfEmptyStrings<T>(rowData:T) {
+    if (rowData instanceof Array) {
+        return Array(rowData.length).join(".").split(".");
+    } else {
+        return [""];
+    }
+}
+
+function getAsList<T>(rowData: T, data: string|string[]) {
+    let length = 1;
+    if (data instanceof Array) {
+        if (!(rowData instanceof Array) || rowData.length != data.length) {
+            throw new Error('Unexpected error!');
```

Comment:
I dont see why unique per row means it needs to be the length of rowData. You could end up outputting more than one row from a single rowData, or fewer rows than the length of rowData. That's the functional reason why I see this approach as wrong. From a code point of view, it violates one of the main rules of this library which is to be completely agnostic as to the form of rowData - once we start going down that road, it becomes less maintainable and general.

I don't even think the length needs to be the same across the row. You can just pad missing entries with the empty string, and build a number of rows equal to the max length column output.